clang/lib/Sema/SemaDecl.cpp

```diff
@@ -9288,7 +9288,8 @@ Sema::ActOnFunctionDeclarator(Scope *S, Declarator &D, DeclContext *DC,
 
     // OpenCL 2.0 pipe restrictions forbids pipe packet types to be non-value
     // types.
-    if (getLangOpts().OpenCLVersion >= 200 || getLangOpts().OpenCLCPlusPlus) {
+    if (getLangOpts().OpenCLVersion >= 200 || getLangOpts().OpenCLCPlusPlus ||
+        getLangOpts().SYCLIsDevice) {
```

Comment:
Do we need `LangOpts` check here?
If I understand correctly, parser won't create `PipeTy` if this check is not `true`, so next check should be enough to cover both conditions. Right?

Reply:
Right. Removed the checks.